### ISSUE TITLE
Question about setting up EndoVis2018. Thank you!

### ISSUE THREAD

**Issue by cekkec** [wenxi-yue/SurgicalSAM#23]
First of all, thank you for your excellent research.
I have some questions regarding the setup of the dataset as mentioned in the README file.

After downloading the dataset through the provided link, the files are organized as shown below. I am wondering how to arrange them according to the File Organization you provided. Could you please provide guidance on this?

In Section 4.1 (Datasets and Evaluation) of the paper, it is mentioned that EndoVis2018 includes 11 training videos and 4 validation videos. However, after downloading the dataset, I found that there are 15 training videos (seq_1 to seq_16, with seq_8 missing) and no separate validation set; instead, there are 4 test sets. Could you clarify whether 4 videos from the 15 training videos were randomly selected for validation, as stated in the paper? Or does the test set refer to the validation set?

I would like to base my research on your excellent work and would greatly appreciate your response.

Thank you in advance for your help.

* The structure below reflects my current setup:
```
SurgicalSAM
|__assets
| ...
|__data
| |__endovis_2018
| | |__train
| | | |__seq1
| | | | |__labels
| | | | | |__frame000.png
| | | | | | ...
| | | | |__left_frames
| | | | | |__frame000.jpg
| | | | | | ...
| | | | |__right_frames
| | | | | |__frame000.jpg
| | | | | | ...
| | | | |__camera_calibration.txt
| | | |
| | | |__seq2
| | | | |...
```

**Comment by wenxi-yue** [comment]
Hi, 

Thank you for your interest. The surgical instrument type annotations for EndoVis2018 were provided by the authors of [ISINet](https://biomedicalcomputervision.uniandes.edu.co/publications/isinet-an-instance-based-approach-for-surgical-instrument-segmentation/), and we follow their data split, which includes 11 training videos and 4 validation videos. You can download their data from their project website, and the file organisation should then make sense. 

Alternatively, you can use our processed data [here](https://unisyd-my.sharepoint.com/:f:/g/personal/wenxi_yue_sydney_edu_au/Et9Nz5d4r2BDkDTakggPMFAB11UNIfbUN3PNz71p1XP0Ug?e=HCYll0).

**Comment by sunshinewuu** [comment]
hi，thank for your great work!I would like to ask if I still need to perform the "Python data_preprocess. py -- dataset endovis_2018-- n-version 40" operation after downloading the dataset from the link you provided?(https://unisyd-my.sharepoint.com/personal/wenxi_yue_sydney_edu_au/_layouts/15/onedrive.aspx?id=%2Fpersonal%2Fwenxi%5Fyue%5Fsydney%5Fedu%5Fau%2FDocuments%2FSurgicalSAM&ga=1) Do we still need data augmentation?

**Comment by wenxi-yue** [comment]
Hi, 

Thanks for your interest. 

Yes, you still need to perform data augmentation. The data provided in this link contains only the original data without any augmentations.

**Comment by prantik-pdeb** [comment]
```
OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq2 
OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq5
OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq9
OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq15
```

All these folders contain 149 images, but all of them are blank images. Could this be a mistake, or is there a specific reason for this?

The folders in `OneDrive_1_12-16-2024\endovis_2018\val\binary_annotations\binary_annotations` (e.g., seq2 or other sequences) have annotations, but the corresponding images are missing in the val folder.

**Comment by wenxi-yue** [comment]
> ```
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq2 
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq5
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq9
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq15
> ```
> 
> All these folders contain 149 images, but all of them are blank images. Could this be a mistake, or is there a specific reason for this?
> 
> The folders in `OneDrive_1_12-16-2024\endovis_2018\val\binary_annotations\binary_annotations` (e.g., seq2 or other sequences) have annotations, but the corresponding images are missing in the val folder.

Hi,

**Comment by wenxi-yue** [comment]
> ```
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq2 
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq5
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq9
> OneDrive_1_12-16-2024\endovis_2018\val\annotations\annotations\seq15
> ```
> 
> All these folders contain 149 images, but all of them are blank images. Could this be a mistake, or is there a specific reason for this?
> 
> The folders in `OneDrive_1_12-16-2024\endovis_2018\val\binary_annotations\binary_annotations` (e.g., seq2 or other sequences) have annotations, but the corresponding images are missing in the val folder.

Hi,

Thank you for your interest.

Regarding question 1 – the annotation images are grayscale, with each pixel representing a class ID. The values range from 0 to 7, which might make them hard to see directly. However, if you load the annotation images as matrices, you should be able to view the class labels.

For question 2 – we provide SAM features for the input images. If you need access to the original images, you can download them from the original datasets. The links are available in our README file.
